src/Microsoft.Azure.SignalR.Protocols/ConnectionMessage.cs

```diff
@@ -158,6 +158,71 @@ public ConnectionDataMessage(string connectionId, ReadOnlySequence<byte> payload
         public ReadOnlySequence<byte> Payload { get; set; }
 
 
+        /// <summary>
+        /// Gets or sets the message ID
+        /// </summary>
+        public ulong? TracingId { get; set; }
+    }
+
+    /// <summary>
+    /// A data message of client invocation completion result.
+    /// </summary>
+    public class ServiceCompletionMessage : ConnectionMessage, IMessageWithTracingId
+    {
+        /// <summary>
+        /// Initialize a new instance of <see cref="ServiceCompletionMessage"/> class.
+        /// </summary>
+        /// <param name="invocationId">The Id of the invocation that has completed.</param>
+        /// <param name="connectionId">The client connection Id that complete the invocation.</param>
+        /// <param name="callerServerId">The serverId that wrap the completion result.</param>
+        /// <param name="payload">The payload of the completion result.</param>
+        /// <param name="tracingId">The tracing Id of the message.</param>
+        public ServiceCompletionMessage(string invocationId, string connectionId, string callerServerId, ReadOnlyMemory<byte> payload, ulong? tracingId = null)
```

Comment:
how server decode payload by the correct protocol, if it initialized by a server without the client.